### DISCUSSION TITLE
Specify an email address to use for commits to an organization's repositories

### DISCUSSION THREAD

**Discussion by garretmh** [community/community#4168] [Pull Requests]
When creating a commit through the GitHub website (e.g. when merging a PR) a user's primary email address is always used. But when working on an organization's code it may be preferable to use an organization-specific email address.

It is already possible to specify an organization-specific email address for notifications and it would be very helpful for GitHub to use that same email address when committing to organization-owned repositories.

**Comment by Levi-Lesches** [comment]
To add some context, this causes major issues in some cases: 

- When the PR is being merged via a squash commit, the email addresses of the committer(s) is lost
- When the PR author doesn't have write-access, the repo manager merges, and _their_ email is recorded
- The above case

**Comment by zomars** [comment]
Also there's more discussion in this issue: https://github.com/todogroup/gh-issues/issues/32

**Comment by sudoforge** [comment]
IMO, this is a gaping flaw and needs to be a high priority issue for the team to resolve.

**Comment by CowChris** [comment]
Yes please!

**Comment by SethAngell** [comment]
Wanted to bump this. Setting a default email address for all web gui actions on a per-organization basis would be highly desirable

**Comment by oshaughnessy** [comment]
Adding another request & data point here. This would be so valuable, and changing the default would probably match expected behavior much more than the current design.

I'll repeat what I just posted in [todogroup's GH Issue 32](https://github.com/todogroup/gh-issues/issues/32#issuecomment-1177943598):

> It has always seemed logical to me that when I join an org, if I have an email address associated with my GitHub account in that org's verified domains, all my PR approvals and merges should start to use that email address. As things stand now, I'm part of an org with 50+ repos, and as part of the devops team, I'm a reviewer or new committer to new repos all the time. I have to maintain constant vigilance when I'm reviewing and merging to be sure I've chosen the right address. I think GitHub could offer a much more sensible default in this scenario.

**Comment by christianbundy** [comment]
The de facto workaround for this is to have all everyone create a new GitHub account specifically for each organization, which is inconvenient and undermines GitHub's goal of individuals having one account. I'd love to be able to advocate for individuals only having one GitHub account, but until this is resolved I'm inclined to recommend that every company makes employees create a GitHub account specifically for work.

**Comment by phlickey** [comment]
i don't want people at work to be able to git blame my personal email address after i retire please github

**Comment by weibelben** [comment]
You should be able to configure an org based custom route under notifications settings 

https://github.com/settings/notifications/custom_routing

**Reply by garretmh** [reply]
This discussion is about using an org email for commits created within GitHub not notifications.

The original post explicitly mentions that this is not about notifications.

**Reply by mscottford** [reply]
That said, perhaps the custom routing that's used for notifications could be expanded to also affect the email that's used for authoring commits that are made via GitHub.

**Reply by ArcanisCz** [reply]
Hi @weibelben custom routing only affects notifications (and for that case it works).

**Comment by zzJinux** [comment]
Closely related: https://github.com/orgs/community/discussions/10840

**Comment by nnutter** [comment]
I noticed today that if I merge I can choose my identity but not if using the merge queue feature.

**Comment by tomgenoni** [comment]
You can add a unique `git` email and name per local repository. 

```
git config user.name "Name Here"
git config user.email "you@company.com"
```

Not a GitHub feature but wouldn't this solve the problem?

**Reply by mjaggard** [reply]
No, it's commits that the GitHub UI makes for you that are the issue, not ones coming from a git push.

**Comment by AmanL04** [comment]
Also facing this issue. Creating a separate GitHub account just to use a different email is a poor workaround. It requires unnecessary local setup and potentially org-level changes. This should be a built-in feature—per-org committer emails for web-based commits are essential for many workflows.

**Comment by Devjeel** [comment]
would love to see this default email setting per organization bases for GitHub PR merge UI

**Comment by alefranz** [comment]
Pretty please! I am so worried one day I will forget to double check I've selected the right email address 😥

I'm sure you don't want to have your users live in constant fear 😅

**Comment by nathipg** [comment]
Looking forward for this feature 🙇🏻‍♀️

**Comment by djl605** [comment]
Come on. It's been _years_ since this issue was opened. This is a huge gap. PLEASE do this.

**Reply by mjaggard** [reply]
When this issue was written GitHub was a platform we could rely on and minor concerns like this were an irritation, now I'd much rather they concentrated on their pathetic [one-9 of uptime ](https://mrshu.github.io/github-statuses/) and don't worry about little bugs like this one.

**Comment by mschoettle** [comment]
Please fix this! It should be possible to specify which commit email to use by default for different organizations same as it is already possible with custom routing.

I thought I could at least simplify it for work by setting my work email as the primary one it seems to default to the first email that is shown under account settings > Emails. :/

**Comment by hsenag** [comment]
Also supporting this feature request!
